src/controllers/customerCommandController.ts

```diff
@@ -0,0 +1,48 @@
+import { ref, watch, Ref } from "vue";
+
+export default function customerCommandController(
+  textInput: Ref<string>,
+  commandsInput: {
+    commandName: string;
+    commandKey: string;
+    commandAction: Function;
+  }[]
+) {
+  const customerCommands: Ref<
+    { commandName: string; commandKey: string; commandAction: Function }[]
+  > = ref([]);
+
+  function getCustomerCommands(
+    textInput: string,
+    commandsInput: {
+      commandName: string;
+      commandKey: string;
+      commandAction: Function;
+    }[]
```

Comment:
Should define `Command` type and export it => other component or end-user could use it